pipeline/src/concluder/mod.rs

```diff
@@ -0,0 +1,307 @@
+            };
+            match batch {
+                Ok(Some(batch)) => {
+                    if batch.num_rows() > 0 {
+                        // Fetch the highest index from the batch
+                        let highest_index = aggregate::max(as_uint64_array(
+                            batch.column_by_name("index").ok_or_else(|| {
+                                anyhow::anyhow!("index column should exist on events record batch")
+                            })?,
+                        )?);
+                        if let Some(highest_index) = highest_index {
+                            last_processed_index = Some(highest_index);
+                        }
+
+                        // Send batch to actor
+                        select! {
+                            _ = &mut shutdown => {
+                                return Ok(());
+                            },
+                            Err(err) = handle.notify(NewEventsMsg { events: batch }) => {
+                                warn!(?err,"failed to notify concluder about new events");
+                            }
+                        };
+                    }
+                }
+                Ok(None) => {
+                    break;
+                }
+                Err(err) => warn!(%err, "failed to poll_next_batch of conclusion_events"),
+            };
+        }
+    }
+}
+
+async fn events_since(
+    ctx: &SessionContext,
+    offset: Option<u64>,
+) -> Result<SendableRecordBatchStream> {
+    // Fetch the conclusion events DataFrame
+    let mut conclusion_events = ctx.table(CONCLUSION_EVENTS_TABLE).await?.select(vec![
+        col("index"),
+        col("stream_cid"),
+        col("stream_type"),
+        col("controller"),
+        col("dimensions"),
+        col("event_cid"),
+        col("event_type"),
+        col("data"),
+        col("previous"),
+    ])?;
+    if let Some(offset) = offset {
+        conclusion_events = conclusion_events.filter(col("index").gt(lit(offset)))?;
+    }
+    Ok(conclusion_events.execute_stream().await?)
+}
+
+/// Request the events since a highwater mark
+#[derive(Debug)]
+pub struct EventsSinceMsg {
+    /// Produce message with an index greater than the highwater_mark.
+    pub highwater_mark: u64,
+}
+impl Message for EventsSinceMsg {
+    type Result = anyhow::Result<SendableRecordBatchStream>;
+}
+
+#[async_trait]
+impl Handler<EventsSinceMsg> for Concluder {
+    async fn handle(&mut self, message: EventsSinceMsg) -> <EventsSinceMsg as Message>::Result {
+        events_since(&self.ctx, Some(message.highwater_mark)).await
+    }
+}
+
+#[async_trait]
+impl StreamTableSource for ConcluderHandle {
+    fn schema(&self) -> SchemaRef {
+        crate::schemas::conclusion_events()
+    }
+    async fn subscribe_since(
+        &self,
+        projection: Option<Vec<usize>>,
+        offset: Option<u64>,
+        limit: Option<usize>,
+    ) -> anyhow::Result<SendableRecordBatchStream> {
+        Ok(self
```

Comment:
Here is where the Concluder implements the `subscribe_since` method on from the source. All it does is send the `SubscribeSinceMsg` to the actor. 

The Aggregator implements the same trait in the same way and registers an `event_states_stream` table. This pattern is repeatable so that any new actors can easily expose a `_stream` suffixed table if needed.